src/test/ui/lint/function-references.rs

```diff
@@ -0,0 +1,22 @@
+// check-pass
+fn foo() -> usize { 42 }
+fn bar(x: usize) -> usize { x }
+fn baz(x: usize, y: usize) -> usize { x + y }
+
+fn main() {
+    println!("{:p}", &foo);
+    //~^ WARN cast `foo` with `as *const fn() -> _` to use it as a pointer
+    println!("{:p}", &bar);
+    //~^ WARN cast `bar` with `as *const fn(_) -> _` to use it as a pointer
+    println!("{:p}", &baz);
+    //~^ WARN cast `baz` with `as *const fn(_, _) -> _` to use it as a pointer
+
+    //should not produce any warnings
+    println!("{:p}", foo as *const fn() -> usize);
```

Comment:
These are raw pointers to function pointers... that is certainly not what we want to test here. Please remove the `*const`.